packages/neuron-ui/src/components/HDWalletSign/index.tsx

```diff
@@ -56,21 +56,12 @@ const HDWalletSign = ({ tx }: { tx: State.DetailedTransaction }) => {
         } else {
           try {
             if (cell.lock.codeHash === systemCodeHash && cell.lock.hashType === 'type') {
-              address = ckbCore.utils.bech32Address(cell.lock.args, {
-                prefix: addressPrefix,
-                type: ckbCore.utils.AddressType.HashIdx,
-                codeHashOrCodeHashIndex: '0x00',
-              })
+              address = generateAddress(cell.lock, { config })
             } else {
-              address = ckbCore.utils.fullPayloadToAddress({
-                args: cell.lock.args,
-                prefix: addressPrefix,
-                type:
-                  cell.lock.hashType === 'data'
-                    ? ckbCore.utils.AddressType.DataCodeHash
-                    : ckbCore.utils.AddressType.TypeCodeHash,
-                codeHash: cell.lock.codeHash,
-              })
+              address = generateAddress(
+                { ...cell.lock, hashType: cell.lock.hashType === 'data' ? 'data' : 'type' },
```

Comment:
Now `hash type` has more than `data`, `type`, so it should be explicitly specified instead of using `type` by default. Error should be thrown if no valid `hash type` is matched.

Ref:
- Valid hash type: https://github.com/nervosnetwork/rfcs/blob/198fc90ab7582953ed85a6655e88e51346857475/rfcs/0051-ckb2023/0051-ckb2023.md?plain=1#L25-L26

Reply:
1. When `hashType` is `data` or `type`, use `generateAddress` to generate deprecated address which matches the original code behavior.
2. When 'hashType' is `data1` or `data2`, use `encodeToAddress` to generate `0x00` format type new full address.
3. Throw an error when `hashType` is invalid.

```ts
try {
  if (cell.lock.codeHash === systemCodeHash && cell.lock.hashType === 'type') {
    address = generateAddress(
      {
        args: cell.lock.args,
        codeHash: config.SCRIPTS.SECP256K1_BLAKE160.CODE_HASH,
        hashType: config.SCRIPTS.SECP256K1_BLAKE160.HASH_TYPE,
      },
      { config }
    )
  } else if (['data', 'type', 'data1', 'data2'].includes(cell.lock.hashType)) {
    address = ['data', 'type'].includes(cell.lock.hashType)
      ? generateAddress(cell.lock, { config })
      : encodeToAddress(cell.lock, { config })
  } else {
    throw new Error(`Invalid lock script hash type ${cell.lock.hashType}`)
  }
} catch (err) {
  console.error(err)
}
```

Reply:
LGTM, but `generateAddress` can be replaced by `encodeToAddress`, because they are equavelant to the same lock script, just in different format.

Reply:
OK. I will commit this.
```ts
try {
  if (cell.lock.codeHash === systemCodeHash && cell.lock.hashType === 'type') {
    address = generateAddress(cell.lock, { config })
  } else if (['data', 'type', 'data1', 'data2'].includes(cell.lock.hashType)) {
    address = encodeToAddress(cell.lock, { config })
  } else {
    throw new Error(`Invalid lock script hash type ${cell.lock.hashType}`)
  }
} catch (err) {
  console.error(err)
}
```